core/src/models/Schedule.ts

```diff
@@ -352,16 +375,17 @@ export class Schedule extends LoggedModel<Schedule> {
     }
   }
 
-  @BeforeCreate
-  static async ensureSourceCanUseSchedule(instance: Schedule) {
-    const source = await Source.findById(instance.sourceId);
-    if (source.state !== "ready") throw new Error("source is not ready");
-
-    const scheduleAvailable = await source.scheduleAvailable();
-    if (!scheduleAvailable) {
-      throw new Error(
-        `source ${source.name} (${instance.sourceId}) cannot have a schedule`
-      );
+  @BeforeSave
+  static async checkIncremental(instance: Schedule) {
+    const { pluginConnection } = await instance.getPlugin();
+    if (instance.incremental && !pluginConnection.supportIncrementalSchedule) {
+      if (instance.isNewRecord) {
+        instance.incremental = false;
+      } else {
```

Comment:
Would this validation be skipped when loading config files for the first time? (it's a new record so it would just be set to false and not throw?)

Reply:
I agree this is strange!
I think this was an artifact of having a incremental=true default value, but really, the UI/action should make sure that you've set a good value.